src/graph/mod.rs

```diff
@@ -463,56 +471,57 @@ impl Graph {
     /// return an index to it.
     pub fn edge_parent<I, J>(&self, idx: I, edge: Edge) -> Option<J>
         where I: GraphIndex,
-              J: GraphIndex,
+              J: GraphIndex
     {
         self.parents(idx).find(self, |g, e, _| g[e] == edge).and_then(|(_, n)| self.convert_idx(n))
     }
 
     /// Return the index of the parent along the given widget's **Depth** **Edge**.
-    pub fn depth_parent<I, J=NodeIndex>(&self, idx: I) -> Option<J>
+    pub fn depth_parent<I, J = NodeIndex>(&self, idx: I) -> Option<J>
         where I: GraphIndex,
-              J: GraphIndex,
+              J: GraphIndex
     {
         self.edge_parent(idx, Edge::Depth)
     }
 
     /// Return the index of the parent along the given widget's **Position** **Edge**.
-    pub fn x_position_parent<I, J=NodeIndex>(&self, idx: I) -> Option<J>
+    pub fn x_position_parent<I, J = NodeIndex>(&self, idx: I) -> Option<J>
         where I: GraphIndex,
-              J: GraphIndex,
+              J: GraphIndex
     {
         self.edge_parent(idx, Edge::Position(Axis::X))
     }
 
     /// Return the index of the parent along the given widget's **Position** **Edge**.
-    pub fn y_position_parent<I, J=NodeIndex>(&self, idx: I) -> Option<J>
+    pub fn y_position_parent<I, J = NodeIndex>(&self, idx: I) -> Option<J>
         where I: GraphIndex,
-              J: GraphIndex,
+              J: GraphIndex
     {
         self.edge_parent(idx, Edge::Position(Axis::Y))
     }
 
     /// Produces an iterator yielding the parents along both the **X** and **Y** **Position**
     /// **Edge**s respectively.
-    pub fn position_parents<I, J=NodeIndex>(&self, idx: I) -> PositionParents<J>
+    pub fn position_parents<I, J = NodeIndex>(&self, idx: I) -> PositionParents<J>
         where I: GraphIndex,
-              J: GraphIndex,
+              J: GraphIndex
     {
         self.x_position_parent(idx).into_iter().chain(self.y_position_parent(idx))
     }
 
     /// Return the index of the parent along the given widget's **Graphic** **Edge**.
-    pub fn graphic_parent<I, J=NodeIndex>(&self, idx: I) -> Option<J>
+    pub fn graphic_parent<I, J = NodeIndex>(&self, idx: I) -> Option<J>
         where I: GraphIndex,
-              J: GraphIndex,
+              J: GraphIndex
     {
         self.edge_parent(idx, Edge::Graphic)
     }
 
     /// A **Walker** type that recursively walks **Depth** parents starting from the given node.
-    pub fn depth_parent_recursion<I: GraphIndex>(&self, idx: I)
-        -> RecursiveWalk<fn(&Graph, NodeIndex) -> Option<IndexPair>>
-    {
+    pub fn depth_parent_recursion<I: GraphIndex>
+                                                 (&self,
+                                                  idx: I)
+                                                  -> RecursiveWalk<fn(&Graph, NodeIndex) -> Option<IndexPair>> {
```

Comment:
I don't quite understand this change, especially the newline between the `>` and the `(`. Do you know why rustmt does this? I see it does this for a few following methods also... Maybe it would look nicer if we just moved the `GraphIndex` bound to a where clause instead? Generally, I think `where` clauses are a bit clearer, and rustfmt seems to have less trouble formatting them nicely.